test/test_lyric.py

```diff
@@ -0,0 +1,397 @@
+import unittest
+from typing import Dict
+from tuneflow_py.models.song import Song
+from tuneflow_py.models.lyric import Lyrics, LyricLine, LyricWord
+
+DEFAULT_PPQ = Song.get_default_resolution()
+
+
+def assert_lyric_lines_equal(expected: Dict, actual: LyricLine):
+    if expected["sentence"] != actual.get_sentence():
+        raise Exception(
+            f'Lyric lines are not equal. Expected sentence {expected["sentence"]} but got {actual.get_sentence()}'
+        )
+    if expected["start_tick"] != actual.get_start_tick():
+        raise Exception(
+            f'Lyric lines are not equal. Expected start tick {expected["start_tick"]} but got {actual.get_start_tick()}'
+        )
+    if expected["end_tick"] != actual.get_end_tick():
+        raise Exception(
+            f'Lyric lines are not equal. Expected end tick {expected["end_tick"]} but got {actual.get_end_tick()}'
+        )
+
+
+def assert_lyric_words_equal(expected: Dict, actual: LyricWord):
+    if expected["word"] != actual.get_word():
+        raise Exception(
+            f'Lyric words are not equal. Expected word {expected["word"]} but got {actual.get_word()}'
+        )
+    if expected["start_tick"] != actual.get_start_tick():
+        raise Exception(
+            f'Lyric words are not equal. Expected start tick {expected["start_tick"]} but got {actual.get_start_tick()}'
+        )
+    if expected["end_tick"] != actual.get_end_tick():
+        raise Exception(
+            f'Lyric words are not equal. Expected end tick {expected["end_tick"]} but got {actual.get_end_tick()}'
+        )
+
+
+def create_lyrics():
+    song = Song()
+    lyrics = Lyrics(song)
+    lyrics.create_line_from_string("Hello world this is a test.", 10, 50)
+    lyrics.create_line_from_string("Mixing symbols & numbers 12345, special #%$^characters, LoWeR uPPeR CaSe", 60, 100)
+    lyrics.create_line_from_string("émphasis on accénts, we also allow overlapping", 50, 110)
+    # This is an example of Chinese lyrics.
+    lyrics.create_line_from_string("这是一个中文歌词的例子。", 500, 550)
+    # This is an example of Japanese lyrics.
+    lyrics.create_line_from_string("これは日本語の歌詞の例です。", 600, 650)
+    return lyrics
+
+
+class TestLyrics(unittest.TestCase):
+    def test_create_line(self):
+        lyrics = create_lyrics()
+        self.assertEqual(len(lyrics), 5)
+        line = lyrics.create_line(660)
+        self.assertEqual(len(lyrics), 6)
+        self.assertEqual(line.is_empty(), True)
+        assert_lyric_lines_equal({"sentence": "", "start_tick": 660, "end_tick": 1140}, line)
+
+    def test_create_line_from_string(self):
+        lyrics = create_lyrics()
+        self.assertEqual(len(lyrics), 5)
+        line = lyrics.create_line_from_string("Hello world! 你好世界！", 10, 50)
```

Comment:
We also need to test the words in it, making sure the tokenizer is working intended and the timespan of the words are assigned correctly.